schematics/types/compound.py

```diff
@@ -151,6 +158,21 @@ def __init__(self, field, min_size=None, max_size=None, **kwargs):
     def model_class(self):
         return self.field.model_class
 
+    def _mock(self, context=None):
+        min_size = self.min_size or 1
+        max_size = self.max_size or 1
+
+        if min_size > max_size:
+            message = u'Minimum list size is greater than maximum list size.'
+            raise MockCreationError(message)
+
+        random_length = get_value_in(min_size, max_size)
+
+        if isinstance(self.field, ModelType):
+            return [self.field.model_class.get_mock_object(context) for _ in xrange(random_length)]
+        else:
```

Comment:
Hmm, are you calling `self.field.model_class.get_mock_object()` directly just to avoid the extra function call via `ModelType._mock()`? If so, then I'd be inclined to always call `self.field._mock()` for simplicity.